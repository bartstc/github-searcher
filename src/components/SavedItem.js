import React from 'react';

const SavedItem = ({ id, full_name, html_url, language, removeRepo }) => {
  const handleRemove = () => removeRepo(id);

  return (
    <li key={id} className="list-group-item mb-4">
      <div className="d-flex justify-content-between align-items-center">
        <a href={html_url} className="link-unstyled"><h6>{full_name}</h6></a>
        <button
          type="button"
          className="btn btn-danger"
          onClick={handleRemove}
        >Delete</button>
      </div>
      <div>
        <span className="badge badge-warning badge-pill mr-3">{language}</span>
      </div>
    </li>
  );
};

export default SavedItem;