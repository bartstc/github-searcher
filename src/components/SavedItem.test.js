import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SavedItem from './SavedItem';

describe('SavedItem', () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = {
    id: 1,
    full_name: 'name',
    html_url: 'url',
    language: 'language',
    removeRepo: mockRemove
  };
  const item = shallow(<SavedItem {...props} />);

  it('renders correctly', () => {
    expect(toJson(item)).toMatchSnapshot();
  });

  describe('when clicking the `Remove` button', () => {
    beforeEach(() => item.find('.btn').simulate('click'));

    it('calls the removeRepo func', () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  });
});