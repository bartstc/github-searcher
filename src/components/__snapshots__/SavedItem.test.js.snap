// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedItem renders correctly 1`] = `
<li
  className="list-group-item mb-4"
  key="1"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <a
      className="link-unstyled"
      href="url"
    >
      <h6>
        name
      </h6>
    </a>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
  <div>
    <span
      className="badge badge-warning badge-pill mr-3"
    >
      language
    </span>
  </div>
</li>
`;
